src/test/java/teammates/test/cases/ui/pagedata/InstructorFeedbackEditPageDataTest.java

```diff
@@ -0,0 +1,335 @@
+                         .withUserId(instructor.googleId)
+                         .withCourseId(fs.courseId)
+                         .withSessionName(fs.feedbackSessionName).toString(),
+                     newQuestionForm.getDoneEditingLink().toString());
+        
+        
+        copyForm = data.getCopyQnForm();
+        assertEquals(0, copyForm.getQuestionRows().size());
+        
+        
+        ______TS("instructor with insufficient permissions");
+        // setup
+        data = new InstructorFeedbackEditPageData(dataBundle.accounts.get("helperOfCourse1"));
+        fs = dataBundle.feedbackSessions.get("session1InCourse1");
+        fs.resultsVisibleFromTime = Const.TIME_REPRESENTS_FOLLOW_VISIBLE;
+        fs.sessionVisibleFromTime = Const.TIME_REPRESENTS_FOLLOW_OPENING;
+        
+        questions = new ArrayList<FeedbackQuestionAttributes>();
+        questions.add(dataBundle.feedbackQuestions.get("qn1InSession1InCourse1"));
+        questions.add(dataBundle.feedbackQuestions.get("qn2InSession1InCourse1"));
+        
+        copiableQuestions = new ArrayList<FeedbackQuestionAttributes>();
+        copiableQuestions.addAll(dataBundle.feedbackQuestions.values());
+        
+        questionHasResponses = new HashMap<String, Boolean>();
+        questionHasResponses.put(dataBundle.feedbackQuestions.get("qn1InSession1InCourse1").getId(), true);
+        
+        studentList = new ArrayList<StudentAttributes>();
+        studentList.add(dataBundle.students.get("student1InCourse1"));
+        
+        instructorList = new ArrayList<InstructorAttributes>();
+        instructorList.add(dataBundle.instructors.get("instructor1OfCourse1"));
+        
+        instructor = getInstructorFromBundle("helperOfCourse1");
+        
+        data.init(fs, questions, copiableQuestions, questionHasResponses, studentList, instructorList, instructor);
+        assertEquals(0, data.getCopyQnForm().getQuestionRows().size());
+    }
+    
+    public InstructorAttributes getInstructorFromBundle(String instructor) {
+        InstructorAttributes instructorAttributes = dataBundle.instructors.get(instructor);
+        instructorAttributes.privileges = gson.fromJson(instructorAttributes.instructorPrivilegesAsText, InstructorPrivileges.class);
+        return instructorAttributes;
+    }
+
+    public void verifyMapContains(Map<String, Boolean> map, List<FeedbackParticipantType> list) {
+        for (FeedbackParticipantType participant : list) {
+            assertTrue(map.get(participant.name()));
+        }
+        Set<FeedbackParticipantType> allParticipantTypes =
+                                        EnumSet.allOf(FeedbackParticipantType.class);
+        Set<String> nonParticipants = new HashSet<String>();
+        for (FeedbackParticipantType participantType : allParticipantTypes) {
+            if (list.contains(participantType.name())) {
+                nonParticipants.add(participantType.name());
```

Comment:
As the list is one of `FeedbackParticipantType`, you should be checking if it contains `participantType`, not `participantType.name()`.
Also, I believe you forgot to negate that `if` condition.

If I'm guessing your objective correctly, this segment of code should be able to do what you're trying for in a more readable manner:
```
Set<FeedbackParticipantType> nonParticipants = new HashSet<>(Arrays.asList(FeedbackParticipantType.values()));
nonParticipants.removeAll(list);
```
This test is only passing right now due to nonParticipants always being empty, though, so you'll have to fix that.